### ISSUE TITLE
[3.1.6] Incorrect HTTPClient registration

### ISSUE THREAD

**Issue by Andrzej-W** [dotnet/AspNetCore.Docs#18359]
On this page and probably in a few other places (for example https://docs.microsoft.com/en-us/aspnet/core/security/blazor/webassembly/additional-scenarios?view=aspnetcore-3.1) HTTPClient is registered as `Transient` which is wrong
```c#
builder.Services.AddTransient(sp => 
    new HttpClient
    {
        BaseAddress = new Uri(builder.HostEnvironment.BaseAddress)
    });
```
See this https://github.com/dotnet/aspnetcore/issues/21655
It looks that project template issue will be addressed in 5.0 but maybe you can make corrections in the doc earlier.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8b5e2e10-ff7b-2b9f-84da-f16028ae2c53
* Version Independent ID: c7e59a08-1c60-32c2-75fd-33cb77ff7a5d
* Content: [Call a web API from ASP.NET Core Blazor WebAssembly](https://docs.microsoft.com/en-us/aspnet/core/blazor/call-web-api?view=aspnetcore-3.1)
* Content Source: [aspnetcore/blazor/call-web-api.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/call-web-api.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
> which is wrong

Not quite "wrong" :smile: ..........

> that project template issue will be addressed in 5.0

Correct, @Andrzej-W ... we can't pull the trigger on *completely changing it* until they patch the current release.

WRT a NOTE or section to address it, perhaps with or without a link to the engineering issue, I'm not opposed to that if they think it serious enough. I'll mark this for triage so Artak can decide what we'll do, if anything.

**Comment by Andrzej-W** [comment]
Default application template is a starting point for developers and it should not contain errors or bad practices. In earlier preview versions HTTPClient was always registered as Singleton. I have to say that I don't understand why it was changed to Transient.

Default application template is usually relatively simple and developers read documentation to find additional examples, recommended solutions and best practices. If there is no reason to register HTTPClinet as Transient it should be clearly stated in the documentation.

> we can't pull the trigger on completely changing it until they patch the current release.

This chapter is not about default application template. I think there is no reason to wait. Fortunately in chapters Overview, Get started, Tutorials and Templates there is no example of HTTPClient registration.

**Comment by guardrex** [comment]
@Andrzej-W Thanks for that feedback, but that's not the paradigm that docs adopt. We don't document the opposite of what engineering says devs should be doing (in this case, in the framework code itself) unless they find a problem serious enough to patch the framework (or at least address it on the next minor release). In my experience, it would be a major break in process to adopt that strategy.

@mkArtakMSFT I haven't received further guidance from the team on this. I'm inclined to close at this time because it isn't currently actionable for docs. In **_recent_** unrelated discussions that involved `HttpClient`, both here and in engineering, no guidance was given to me to change the registration. This is fundamentally still an engineering issue at this point. We probably can't tell devs **_not to do something in the text_** that the template preconfigures in their apps. I mean if there's a patch coming soon, ok; but if there isn't a patch (or next minor release update perhaps), it doesn't seem to make sense to take action at this time. We can re-open this issue or create a new issue when the time comes to make changes. Having unactionable docs issues lying around is a bad UE .... **_for ME._** :smile: lol How would u like to proceed?

**Comment by guardrex** [comment]
We're just waiting on 3.1.6 to land, and I'll make the updates.

https://github.com/dotnet/aspnetcore/pull/22770

**Comment by guardrex** [comment]
Cool beans ... **_3.1.6 is out now_**, so I'll work this first thing Thursday (7/16) morning.
